### PR TITLE
phc: add new metric to counter number of unique requests mitigated

### PR DESCRIPTION
**Pull request by MustafaSaber**
related to https://github.com/zalando/skipper/pull/3035
depends on https://github.com/zalando/skipper/pull/3058

### CODE REVIEW

File: docs/operation/operation.md [hunk 1]
```diff
@@ -932,6 +932,7 @@ while choosing the endpoint for the given request
 A set of metrics will be exposed to track passive health check:
 
 * `passive-health-check.endpoints.dropped`: Number of all endpoints dropped before load balancing a request, so after N requests and M endpoints are being dropped this counter would be N*M.
+* `passive-health-check.requests.passed`: Number of unique requests where PHC was able to avoid sending them to unhealthy endpoints.
```
Comment (RomanZavodskikh) [inline docs/operation/operation.md:935]: minor: you can link it to numbers mentioned in previous metrics and tell it's 'N' actually.

**Comment by MustafaSaber** [comment]
Metrics will differ when we have more than one unhealthy endpoint that's why we landed https://github.com/zalando/skipper/pull/3058 first

File: proxy/healthy_endpoints_test.go [hunk 2]
```diff
@@ -197,6 +197,7 @@ func TestPHCForMultipleHealthyAndOneUnhealthyEndpoints(t *testing.T) {
 			assert.InDelta(t, 0, float64(failedRequests), 0.1*float64(va.TotalRequests()))
 			mockMetrics.WithCounters(func(counters map[string]int64) {
 				assert.InDelta(t, float64(va.TotalRequests()), float64(counters["passive-health-check.endpoints.dropped"]), 0.3*float64(va.TotalRequests())) // allow 30% error
+				assert.InDelta(t, float64(nRequests), float64(counters["passive-health-check.requests.passed"]), 0.3*float64(nRequests))                     // allow 30% error
```
Comment (RomanZavodskikh) [inline proxy/healthy_endpoints_test.go:200]: Should it be `va.TotalRequests()` here, just like on previous line?

File: proxy/healthy_endpoints_test.go [hunk 3]
```diff
@@ -251,6 +254,7 @@ func TestPHCForMultipleHealthyAndMultipleUnhealthyEndpoints(t *testing.T) {
 			assert.InDelta(t, 0, float64(failedRequests), 0.3*float64(va.TotalRequests()))
 			mockMetrics.WithCounters(func(counters map[string]int64) {
 				assert.InDelta(t, 2*float64(va.TotalRequests()), float64(counters["passive-health-check.endpoints.dropped"]), 0.6*float64(va.TotalRequests()))
+				assert.InDelta(t, float64(nRequests), float64(counters["passive-health-check.requests.passed"]), 0.3*float64(nRequests)) // allow 30% error
```
Comment (RomanZavodskikh) [inline proxy/healthy_endpoints_test.go:257]: Should it be `va.TotalRequests()` here, just like on previous line?

File: proxy/healthy_endpoints_test.go (hunk 2 above)
Comment (MustafaSaber) [reply proxy/healthy_endpoints_test.go]: we are lucky it worked because `nRequests == va.TotalRequests()`

File: docs/operation/operation.md (hunk 1 above)
Comment (MustafaSaber) [reply docs/operation/operation.md:935]: I think adding the variable here will make a confusion.

**Comment by RomanZavodskikh** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**